File: platforms/flutter/ios/Classes/RaTeXPlugin.swift
import Flutter
import UIKit

/// Minimal Flutter plugin registration for ratex_flutter.
///
/// All rendering happens via Dart FFI (DynamicLibrary.process()) — the symbols
/// from libratex_ffi are linked into the app binary through the xcframework.
/// No method channels or event channels are needed.
public class RaTeXPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        // FFI-only plugin: no channels to register.
    }
}

File: platforms/flutter/android/src/main/kotlin/io/ratex/RaTeXPlugin.kt
// RaTeXPlugin.kt — Flutter plugin registration for ratex_flutter (Android).
//
// All rendering is done via Dart FFI: DynamicLibrary.open("libratex_ffi.so").
// The .so is packaged from android/jniLibs (see build.gradle sourceSets;
// run platforms/android/build-android.sh to build libratex_ffi.so).
// No method channels or event channels.

package io.ratex

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Minimal Flutter plugin for RaTeX — no platform channels. */
class RaTeXPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // FFI-only: Dart loads libratex_ffi.so via DynamicLibrary.open.
        // No registration of method channels.
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
